### PR TITLE
Feature/resend activation code

### PR DESCRIPTION
**Pull request by mohamedebrahim2020**

### CODE REVIEW

**Comment by MohannadElemary2** [comment]
Please resolve conflicts
@mohamedebrahim2020

File: app/Http/Controllers/AppointmentController.php [hunk 1]
```diff
@@ -46,4 +46,9 @@ public function reject(RejectAppointmentRequest $request, $id)
         $appointment = $this->service->reject($request->except('status'), $id);
         return response()->json(new UpdatedResource($appointment), Response::HTTP_OK);        
     }
+
+    public function test()
+    {
+        dd('hima');
+    }
```
Comment (MohannadElemary2) [inline app/Http/Controllers/AppointmentController.php]: Delete this

File: app/Http/Requests/ResendVerificationCodeRequest.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'email' => 'required|email|exists:patients,email',
+            'password' => 'required',
```
Comment (MohannadElemary2) [inline app/Http/Requests/ResendVerificationCodeRequest.php]: The user will not send his password in resend code

File: app/Filters/PatientVerificationCodeFilters.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php
+namespace App\Filters;
+
+use Illuminate\Http\Request;
+
+class PatientVerificationCodeFilters extends QueryFilters
+{    
+    public function email() 
+    {
+        return $this->builder->where('deleted_at', null);
```
Comment (MohannadElemary2) [inline app/Filters/PatientVerificationCodeFilters.php]: When you use SoftDeletes trait in the model this condition applies by default

File: app/Repositories/PatientVerificationCodeRepository.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    * @param Patient $model
+    */
+   public function __construct(PatientVerificationCode $model)
+   {
+       parent::__construct($model);
+   }
+
+   public function fiterPatientCode($patient)
+   {
+        $code = $patient->verificationCode()->filter(app(PatientVerificationCodeFilters::class))->get();
```
Comment (MohannadElemary2) [inline app/Repositories/PatientVerificationCodeRepository.php]: Just delete them
`$patient->verificationCode()->delete();`

File: app/Services/PatientService.php [hunk 5]
```diff
@@ -74,4 +75,16 @@ public function checkDuration($shift, $duration)
             }
         } 
     }
+
+    public function codeResend($data)
+    {
+        $patient = $this->repository->findPatientByEmail($data);
+        (!Hash::check($data['password'], $patient->password)) ? abort(Response::HTTP_UNAUTHORIZED) : "" ;
+        ($patient->verified_at) ? abort(Response::HTTP_BAD_REQUEST) : "" ;
```
Comment (MohannadElemary2) [inline app/Services/PatientService.php]: Use `full if condition` if you're not going to return anything in a variable

File: app/Services/PatientVerificationCodeService.php [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        $this->repository = $repository;
+    }
+
+    public function deletePatientOldCode($patient)
+    {
+        $code = $this->repository->fiterPatientCode($patient)->toArray();
+        if ($code) {
+           $this->delete($this->show($code[0]['id'])); 
+        }
```
Comment (MohannadElemary2) [inline app/Services/PatientVerificationCodeService.php]: Use the approach I've commented in the repo and delete this

File: app/Http/Controllers/AppointmentController.php (hunk 1 above)
Comment (mohamedebrahim2020) [reply app/Http/Controllers/AppointmentController.php]: done

File: app/Http/Requests/ResendVerificationCodeRequest.php (hunk 2 above)
Comment (mohamedebrahim2020) [reply app/Http/Requests/ResendVerificationCodeRequest.php]: done

File: app/Filters/PatientVerificationCodeFilters.php (hunk 3 above)
Comment (mohamedebrahim2020) [reply app/Filters/PatientVerificationCodeFilters.php]: done

File: app/Repositories/PatientVerificationCodeRepository.php (hunk 4 above)
Comment (mohamedebrahim2020) [reply app/Repositories/PatientVerificationCodeRepository.php]: done

File: app/Services/PatientService.php (hunk 5 above)
Comment (mohamedebrahim2020) [reply app/Services/PatientService.php]: done

File: app/Services/PatientVerificationCodeService.php (hunk 6 above)
Comment (mohamedebrahim2020) [reply app/Services/PatientVerificationCodeService.php]: done

**Pull request merged**